raiden/ui/app.py

```diff
@@ -258,74 +175,12 @@ def run_app(
-    given_network_id = network_id
-    node_network_id = int(web3.version.network)  # pylint: disable=no-member
-    known_given_network_id = given_network_id in ID_TO_NETWORKNAME
-    known_node_network_id = node_network_id in ID_TO_NETWORKNAME
-
-    if node_network_id != given_network_id:
-        if known_given_network_id and known_node_network_id:
-            click.secho(
-                f"The chosen ethereum network '{ID_TO_NETWORKNAME[given_network_id]}' "
-                f"differs from the ethereum client '{ID_TO_NETWORKNAME[node_network_id]}'. "
-                "Please update your settings.",
-                fg='red',
-            )
-        else:
-            click.secho(
-                f"The chosen ethereum network id '{given_network_id}' differs "
-                f"from the ethereum client '{node_network_id}'. "
-                "Please update your settings.",
-                fg='red',
-            )
-        sys.exit(1)
+    node_network_id, known_node_network_id = setup_network_id_or_exit(config, network_id, web3)
 
-    config['chain_id'] = given_network_id
-
-    # interpret the provided string argument
-    if environment_type == Environment.PRODUCTION:
-        # Safe configuration: restrictions for mainnet apply and matrix rooms have to be private
-        config['environment_type'] = Environment.PRODUCTION
-        config['transport']['matrix']['private_rooms'] = True
-    else:
-        config['environment_type'] = Environment.DEVELOPMENT
-
-    environment_type = config['environment_type']
-    print(f'Raiden is running in {environment_type.value.lower()} mode')
+    environment_type = setup_environment(config, environment_type)
 
     chain_config = {}
-    contract_addresses_known = False
-    contracts = dict()
-    services_contracts = dict()
-
-    if environment_type == Environment.DEVELOPMENT:
-        contracts_version = DEVELOPMENT_CONTRACT_VERSION
-    else:
-        contracts_version = RED_EYES_CONTRACT_VERSION
-
-    config['contracts_path'] = contracts_precompiled_path(contracts_version)
-
-    if node_network_id in ID_TO_NETWORKNAME and ID_TO_NETWORKNAME[node_network_id] != 'smoketest':
-        deployment_data = get_contracts_deployment_info(
-            chain_id=node_network_id,
-            version=contracts_version,
-        )
-        not_allowed = (  # for now we only disallow mainnet with test configuration
-            network_id == 1 and
-            environment_type == Environment.DEVELOPMENT
-        )
-        if not_allowed:
-            click.secho(
-                f'The chosen network ({ID_TO_NETWORKNAME[node_network_id]}) is not a testnet, '
-                'but the "development" environment was selected.\n'
-                'This is not allowed. Please start again with a safe environment setting '
-                '(--environment production).',
-                fg='red',
-            )
-            sys.exit(1)
-
-        contracts = deployment_data['contracts']
```

Comment:
This broke the filter installation. Now `start_block` will always be `0` here:

https://github.com/raiden-network/raiden/blob/8797ec58a776d992b18ecd26f7788926546b2eac/raiden/ui/app.py#L252-L255